File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/state/AfterNoonState.java
package jv.com.tencent.cain.design_pattern.state;

/**
 * @author cainjiang
 * @date 2020/3/27
 */
public class AfterNoonState extends State {
    @Override
    public void doWork(Work work) {
        if (work.getHour() < 17) {
            System.out.println("当前时间：" + work.getHour() + ", 属于下午工作时间");
        } else {
            work.setState(new EveningState());
            work.doWork();
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/state/AfterNoonState.kt
package kt.com.tencent.cain.desigin_pattern.state

/**
 * @author : jiangyu
 * @date   : 2020/3/28
 * @desc   : xxx
 */
class AfterNoonState : State() {
    override fun doWork(work: Work) {
        if (work.curTimeHour < 19) {
            println("当前时间：${work.curTimeHour}点，属于下午上班时间")
        } else {
            work.state = EveningState()
            work.doWork()
        }
    }
}